File: javascript/day1.js
import fs from 'fs'

const groups = fs
  .readFileSync('../inputs/day1.txt')
  .toString('utf-8')
  .split('\n\n')

const getSumOfGroup = (group) =>
  group
    .split('\n')
    .map(Number)
    .reduce((sum, num) => sum + num, 0)

const sortedGroupSums = groups.map(getSumOfGroup).sort((a, b) => b - a)

const part1 = sortedGroupSums[0]
const part2 = sortedGroupSums.slice(0, 3).reduce((a, b) => a + b, 0)

console.log(part1)
console.log(part2)

File: python/day1.py
input = open("../inputs/day1.txt").readlines()

n, Elves = 0, [0]

for line in map(lambda x: x.strip(), input):
    if line:
        Elves[n] += int(line)
    else:
        n, Elves = n + 1, Elves + [0]


def part1():
    return max(Elves)


def part2():
    return sum(sorted(Elves)[-3:])


if __name__ == "__main__":
    print(part1())
    print(part2())
